### DISCUSSION TITLE
Rotary encoder as a volume control

### DISCUSSION THREAD

**Discussion by dukeplay** [rern/rAudio#621] [Feature Requests]
Hello, first of all, thank you very much for the great software. I have one more request until it has matured to perfection for me. I would like to install a KY-040 rotary control to control the volume. The push button is to be assigned the toggle function. Would it be possible to upgrade this feature? We are also happy to provide instructions on how to manually integrate the rotary control. The instructions found on the Internet are not really promising. Thanks very much.

**Comment by rern** [comment]
Please try the rotary encoder in today's update.

**Reply by dukeplay** [reply]
Hello rern, the rotary knob works immediately and with great accuracy. Thank you very much and Merry Christmas!

**Reply by pepper7777** [reply]
In my case there is noticeable delay when adjusting volume.
Is there anything I can do to improve responsiveness please?

thanks
Pepper

**Comment by Bautar** [comment]
> Hello, first of all, thank you very much for the great software. I have one more request until it has matured to perfection for me. I would like to install a KY-040 rotary control to control the volume. The push button is to be assigned the toggle function. Would it be possible to upgrade this feature? We are also happy to provide instructions on how to manually integrate the rotary control. The instructions found on the Internet are not really promising. Thanks very much.

Hello Dukeplay - I’m an intermediate level RAudio user & interested in adding a rotary encoder to one of my setups. You indicate you may have instructions on manually integrating & I’d love to see some details, particularly I’m using a Raspberry Pi 3B with a HiFiBerryDigi+ card. 

Thanks in advance!
Bautar

**Reply by dukeplay** [reply]
Hello Bautar, sorry for the late reply. I use a standard KY040 rotary encoder. I currently have an IQauidio DAC+ but have already worked with Hifiberry. Hifiberry has a good documentation of the occupied pins by the DAC. Then you need a GPIO PIN header 1 to 2 to make the PINS accessible again. Then you need three free PINS that are not used by the DAC or by other devices. For me it was PIN 15 (CLK), 13 (DT) and 11 (SW). Of course still ground and 3.3V. The software adaptation has been perfectly implemented by rern and integrated into rAudio. Just select the PINS and that's it. Attention PIN is not the same as GPIO! Have you figured it out and implemented it yourself? Greetings Dukeplay
